src/ERC1967Proxy.ts

```diff
@@ -1,14 +1,27 @@
 import { ponder } from "@/generated";
-import { bytesToString, toBytes } from "viem";
+import { bytesToString, parseAbi, toBytes } from "viem";
+import { client } from "./client";
+
+const abi = parseAbi(['function idOf(address) view returns (uint256)']);
+const address = '0xda107a1caf36d198b12c16c7b6a1d1c795978c42';
 
 ponder.on("ERC1967Proxy:Transfer", async ({ event, context }) => {
   const { Fname } = context.entities;
+
+  const fid = await client.readContract({
```

Comment:
Ponder natively supports read-only contract calls via the `context.contracts` object. These magical contract instances implement caching to speed up indexing on subsequent hot reloads/redeploys. They also automatically sets the `blockNumber: event.block.number`.  https://ponder.sh/api-reference/event-handlers#readonlycontract

```ts
const fid = await context.contracts.IdRegistry.read.idOf([event.params.to])
```

So, you don't need to create a `client` as you've done here. The provided contract instances are already hooked up to a viem `PublicClient` that Ponder manages internally.

Reply:
oh wow this is great, TIL!